src/machines/java/com/enderio/machines/common/io/fluid/MachineFluidTank.java

```diff
@@ -0,0 +1,250 @@
+package com.enderio.machines.common.io.fluid;
+
+import com.enderio.EnderIO;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BucketItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraftforge.common.capabilities.ForgeCapabilities;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+
+public class MachineFluidTank extends FluidTank {
+
+    public static final int DEFAULT_CAPACITY = 8 * FluidType.BUCKET_VOLUME;
```

Comment:
What is the rationale behind a default capacity? Surely we can just expect a capacity to be provided always?

Reply:
The rationale is that we probably want a default unified size for all machines, we wouldn't want the vat to have tanks with 16 buckets size, a combustion generator with 5, and other machines with other random numbers added at different points in time. We would like it to be set to one default value and then if there is a specific reason you may change it to change a specific need of a specific machine. Now I removed the field and just inlined it into one of the constructors.